R/build-reference.R

```diff
@@ -6,8 +6,8 @@
 #'
 #' @section YAML config:
 #' To tweak the index page, add a section called `reference` to `_pkgdown.yml`
-#' which provides a list of sections containing, a `title`, list of `contents`,
-#' and an optional `description`.
+#' which provides a list of sections containing a `title` or `subtitle`,
+#' list of `contents`, and an optional `desc`ription.
 #'
```

Comment:
Should provide a concrete example of a YAML spec for a reference with / without subtitles. Something like the following (does this YAML look valid?):

The [keras](https://github.com/rstudio/keras/blob/master/pkgdown/_pkgdown.yml) reference can be organized with by multiple `title`s.

``` yaml
reference:
- title: "Locally-connected Layers"
  contents:
    - layer_locally_connected_1d
    - layer_locally_connected_2d
- title: "Recurrent Layers"
  contents:
    - layer_simple_rnn
    - layer_gru
```
But "Layers" is a natural grouping to create multiple `subtitle`s under a Layer `title`:

``` yaml
reference:
- title: Layers
  desc: Layer types
  subtitle: Locally-connected Layer
  contents:
    - layer_locally_connected_1d
    - layer_locally_connected_2d
  subtitle: Recurrent Layer
  contents:
    - layer_simple_rnn
    - layer_gru
```